[dist_optim] update the doc of DistributedOptimizer

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51314 [dist_optim] update the doc of DistributedOptimizer**

updating the doc of DistributedOptimizer to include TorchScript enablement information

Differential Revision: [D26156032](https://our.internmc.facebook.com/intern/diff/D26156032)

Commit:
[dist_optim] update the doc of DistributedOptimizer

updating the doc of DistributedOptimizer to include TorchScript enablement information

[ghstack-poisoned]

Commit:
Update on "[dist_optim] update the doc of DistributedOptimizer"


updating the doc of DistributedOptimizer to include TorchScript enablement information

[ghstack-poisoned]

rohan-varma commented on torch/distributed/optim/optimizer.py:
```diff
@@ -150,6 +150,13 @@ class DistributedOptimizer:
     to the latest forward pass executed on a given worker. Also, there is no
     guaranteed ordering across workers.
 
+    DistributedOptimizer creates the local optimizer with TorchScript enabled
```
Nit: backticks for DistributedOptimizer

Commit:
Update on "[dist_optim] update the doc of DistributedOptimizer"


updating the doc of DistributedOptimizer to include TorchScript enablement information

[ghstack-poisoned]
